pinot-core/src/main/java/com/linkedin/pinot/server/realtime/ServerSegmentCompletionProtocolHandler.java

```diff
@@ -33,13 +35,37 @@
   private static Logger LOGGER = LoggerFactory.getLogger(ServerSegmentCompletionProtocolHandler.class);
   private static final int SEGMENT_UPLOAD_REQUEST_TIMEOUT_MS = 30_000;
   private static final int OTHER_REQUESTS_TIMEOUT = 5_000;
+  private static final String HTTPS_PROTOCOL = "https";
+
+  private static String _preferredProtocol = null;
+  private static Integer _preferredPort = null;
 
   private final String _instanceId;
   private final FileUploadDownloadClient _fileUploadDownloadClient;
+  private static SSLContext _sslContext;
+
+  public static void setPreferredProtocol(@Nonnull String _preferredProtocol) {
+    ServerSegmentCompletionProtocolHandler._preferredProtocol = _preferredProtocol.toLowerCase();
+    if (HTTPS_PROTOCOL.equals(_preferredProtocol)) {
+      try {
+        _sslContext = new ClientSSLContextGenerator.Builder().build().generate();
```

Comment:
??? Empty SSLContext?

Reply:
No, this is the default, which is "accept all server certs" and "do not present any client certs"